src/test/java/org/apache/commons/collections4/bloomfilter/AbstractIndexProducerTest.java

```diff
@@ -93,8 +98,36 @@ int[] toArray() {
      */
     protected abstract int getBehaviour();
 
+    /**
+     * Creates an array of expected indices.
+     * @return an array of expected indices.
+     */
+    protected abstract int[] getExpectedIndices();
```

Comment:
Note that this method requires assumptions. The expected indices are dependent on the producer returned from createProducer, which is ultimately dependent on the shape of the Bloom filter.